src/server/middleware/auth.ts

```diff
@@ -165,18 +165,24 @@ export const onRequest = async ({
       const payload = decoded.payload as JwtPayload;
       const header = decoded.header;
 
-      // Get the public key from the `iss` payload field or `kid` header field.
-      const publicKey = payload.iss ?? header.kid;
+      // Get the public key from the `iss` payload field.
+      const publicKey = payload.iss;
       if (publicKey) {
         const authWallet = await getAuthWallet();
         if (publicKey === (await authWallet.getAddress())) {
           return await handleAccessToken(jwt, req, getUser);
         } else if (publicKey === THIRDWEB_DASHBOARD_ISSUER) {
           return await handleDashboardAuth(jwt);
         } else {
-          return await handleKeypairAuth(jwt, req, publicKey);
+          return await handleKeypairAuth({ jwt, req, publicKey });
         }
       }
+
+      // Get the public key hash from the `kid` header.
```

Comment:
Breaking change (though undocumented): The "kid" header accepts the hash instead of plaintext public key.